### PR TITLE
refactor: adjustments with checkout, addtocart, removefromcart, updatecart methods for  coupon code

### PR DESCRIPTION
**Pull request by aureate-labs-team**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/api-client/src/api/getProduct/index.ts [hunk 1]
```diff
@@ -125,9 +125,10 @@ export async function getProduct(
         }
       }
     }`
+    const curLocaleCode = params.curLocaleCode;
     const variables = {
       handle: params.slug,
-      country: getCountry(context),
+      country: curLocaleCode === "en" ? "US" : (curLocaleCode).toUpperCase(),
```
Comment (Mateuszpietrusinski) [inline packages/api-client/src/api/getProduct/index.ts:131]: This is a bigger discussion on how to handle correctly the countries. We use configuration for i18n. For example:
```
{
  'en-US': ...something,
  'en-UK': ...something,
  'en-AU': ...something
}

Then you need to take the country from the code or maybe the i18n library somehow exposes the current country base on that format. Its better becouse we are not forced to hardcode the country like here

File: packages/composables/src/getters/cartGetters.ts [hunk 2]
```diff
@@ -88,10 +88,8 @@ export const getCartTotalDiscount = (cart: Cart): number => {
 };
 
 export const getAppliedCoupon = (cart: Cart): string => {
-  if (cart?.checkoutUserErrors?.length <= 0) {
-    return cart.couponCode;
-  }
-  return '';
+  if (cart?.discountApplications.length > 0) 
+  return cart.discountApplications[0].code;
```
Comment (Mateuszpietrusinski) [inline packages/composables/src/getters/cartGetters.ts:92]: We are expecting from that function to return `string` not `void`, after the changes if discount doesn't exist we are not returning any value

File: packages/theme/components/CartSidebar.vue [hunk 3]
```diff
@@ -101,12 +101,16 @@
...
             </template>
               <template #value>
                 <SfPrice
-                  :regular="$n(totalDiscount.percentage ? totalDiscount.percentage/100 : totalDiscount.amount, totalDiscount.percentage ? 'percent':'currency')"
+                  v-if="!totalDiscount.percentage"
+                  :regular="$n(totalDiscount.amount, 'currency')"
                 />
+                <SfPrice
+                v-else
+                :regular="$n(totals.subtotal * totalDiscount.percentage/ (100 - totalDiscount.percentage), 'currency')"/>
```
Comment (Mateuszpietrusinski) [inline packages/theme/components/CartSidebar.vue:113]: this if is big maybe we can move the logic to the script part and here just return the value

File: packages/theme/pages/articles/_handle.vue [hunk 4]
```diff
@@ -40,7 +40,9 @@ export default defineComponent({
       await search({
         contentType: ContentType.Article,
         id: route.value.query.id as string
-      });
+      }).then(()=>
+        console.log('contents::', article.value)
```
Comment (Mateuszpietrusinski) [inline packages/theme/pages/articles/_handle.vue:44]: should we just console log that value?

**Review by Mateuszpietrusinski — CHANGES_REQUESTED** [review]
Some questions and adjustment needed

File: packages/api-client/src/api/getProduct/index.ts (hunk 1 above)
Comment (aureate-labs-team) [reply packages/api-client/src/api/getProduct/index.ts:131]: Will be resolved by PR #262

File: packages/composables/src/getters/cartGetters.ts (hunk 2 above)
Comment (aureate-labs-team) [reply packages/composables/src/getters/cartGetters.ts:92]: Will be resolved by PR #262

File: packages/theme/components/CartSidebar.vue (hunk 3 above)
Comment (aureate-labs-team) [reply packages/theme/components/CartSidebar.vue:113]: It will be resolved by PR #262

File: packages/theme/pages/articles/_handle.vue (hunk 4 above)
Comment (aureate-labs-team) [reply packages/theme/pages/articles/_handle.vue:44]: It will be resolved by PR #262

**Review by jeffpdotone — APPROVED** [review]

**Pull request merged**